robot_calibration/src/optimizer.cpp

```diff
@@ -64,22 +66,71 @@ int Optimizer::optimize(OptimizationParams& params,
     return -1;
   }
 
+  std::stringstream      segments_out;
+  const KDL::SegmentMap& segments = tree_.getSegments();
+  segments_out << "Segment name in kdl tree: " << std::endl;
+  for (KDL::SegmentMap::const_iterator it = segments.cbegin(); it != segments.cend(); it++)
```

Comment:
`std::cbegin()` is `c++14`...this does not compile under `xenial`...
or you have to use `vector::cbegin`....i.e. add `#include <vector>`?

same with `std::cend()`

Reply:
Member function```std::vector::cbegin()``` is C++11 and the global function ```std::cbegin``` is C++14, yes. 
Wiil change it to C++98 mutable iterator ```std::vector::begin```.